microsoft/botframework-solutions
# [ToDo] Optimize flow for first-time user based on new design

Optimize flow for first-time user based on new design
## Description

- When users first triggers ToDo skill, they will touch below steps:
    1. First step is to pop up a message to tell user that we are creating three types of lists(ToDo/Shopping/Grocery) for them automatically. Please keep patience to wait. For OneNote, we will create one Notebook, one section and three pages. For Outlook, we will create three task folders in Tasks tab.
     2. After created, the skill will send an email to users with the logged account. For OneNote, the message contains the online url of the created Notebook. For Outlook, the message contains the official url of the Outlook Tasks.
    3. Last step is to pop a message to tell users that we have created three lists and sent the email with links to them. They can note down their to dos now.

- Store the page ids or task folder ids into user state. In this way, we can maintain the task pages or folders per user. When users trigger the ToDo skill next time, we can use those ids directly to access the OneNote or task folders instead of searching them by http calls which is time-consuming.

## Related Issue
closes #459

## Testing Steps
1. Use a new account to trigger todo skill. After it creates pages for you, you will get a notification email from your logged account.
2. Restart the conversation, you will not see any setting up OneNote/Outlook messages again.

## Checklist
